training_controller/training_controller.py

```diff
@@ -0,0 +1,320 @@
+# Standard Library
+import asyncio
+import json
+import logging
+import os
+import signal
+import time
+
+# Third Party
+import kubernetes.client
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from nats.aio.client import Client as NATS
+from prepare_training_logs import PrepareTrainingLogs
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(name)s - %(message)s")
+config.load_incluster_config()
+configuration = kubernetes.client.Configuration()
+api_instance = kubernetes.client.BatchV1Api()
+logging.info("Cluster config has been been loaded")
+MINIO_ACCESS_KEY = os.environ["MINIO_ACCESS_KEY"]
+MINIO_SECRET_KEY = os.environ["MINIO_SECRET_KEY"]
+NATS_SERVER_URL = os.environ["NATS_SERVER_URL"]
+nulog_spec = {
+    "name": "nulog-train",
+    "container_name": "nulog-train",
+    "image_name": "amartyarancher/nulog-train",
+    "image_pull_policy": "Always",
+    "labels": {"app": "nulog-train"},
+    "restart_policy": "Never",
+    "requests": {"memory": "1Gi", "cpu": 1},
+    "limits": {"nvidia.com/gpu": 1},
+    "env": [],
+}
+nulog_spec["env"] = [
+    client.V1EnvVar(name="MINIO_ACCESS_KEY", value=MINIO_ACCESS_KEY),
+    client.V1EnvVar(name="MINIO_SECRET_KEY", value=MINIO_SECRET_KEY),
+    client.V1EnvVar(name="NATS_SERVER_URL", value=NATS_SERVER_URL),
+]
+startup_time = time.time()
+
+
+def job_not_currently_running(job_name, namespace="default"):
+    try:
+        jobs = api_instance.list_namespaced_job(
+            namespace, pretty=True, timeout_seconds=60
+        )
+    except ApiException as e:
+        logging.error(
+            "Exception when calling BatchV1Api->list_namespaced_job: %s\n" % e
+        )
+
+    for job in jobs.items:
+        if job.metadata.name == job_name:
+            return False
+    return True
+
+
+def check_training_job(job_name, namespace="default"):
+    try:
+        jobs = api_instance.list_namespaced_job(
+            namespace, pretty=True, timeout_seconds=60
+        )
+    except ApiException as e:
+        logging.error(
+            "Exception when calling BatchV1Api->list_namespaced_job: %s\n" % e
+        )
+
+    for job in jobs.items:
+        if job.metadata.name == job_name:
+            return False
+    return True
```

Comment:
`job_not_currently_running` and `check_training_job` appear to be doing the same thing - remove 1 of them